src/main/java/duke/module/Lesson.java

```diff
@@ -0,0 +1,132 @@
+package duke.module;
+
+import duke.data.Storage;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.Map;
+
+public class Lesson {
+    /**
+     * A hash map which holds the optional lessons learnt for any day.
+     */
+    private Map<Date, ArrayList<String>> lessons;
+
+    /**
+     * Constructor for Lesson objects.
+     * @param dateArrayListMap The hash map of lessons learnt for the day from
+     *                         loading the lessons.txt text file.
+     */
+    public Lesson(final Map<Date, ArrayList<String>> dateArrayListMap) {
+        this.lessons = dateArrayListMap;
+    }
+
+    /**
+     * Shows all lessons learnt for a specific day.
+     * @param day The day to view all lessons learnt.
+     * @return A message containing all the lessons learnt for the day
+     * to be printed.
+     * @throws ParseException if the user input is in wrong format.
+     */
+    public String viewLesson(final String day) throws ParseException {
+        SimpleDateFormat simpleDateFormat = new SimpleDateFormat("dd/MM/yyyy");
+        Date today = simpleDateFormat.parse(day);
+        String message = "";
+        boolean hasLesson = false;
+        for (Date d : lessons.keySet()) {
+            if (d.equals(today)) {
+                if (!lessons.get(d).isEmpty()) {
+                    hasLesson = true;
+                    for (String str : lessons.get(d)) {
+                        message += str + "\n";
+                    }
+                }
+            }
+        }
```

Comment:
Convoluted and unnecessary nesting. Look through the Map documentation and you'll find a better way to do this.